thunx-encoding-json/src/main/java/com/contentgrid/thunx/encoding/json/JsonCollectionValueDto.java

```diff
@@ -0,0 +1,84 @@
+package com.contentgrid.thunx.encoding.json;
+
+import com.contentgrid.thunx.predicates.model.CollectionValue;
+import com.contentgrid.thunx.predicates.model.ListValue;
+import com.contentgrid.thunx.predicates.model.Scalar;
+import com.contentgrid.thunx.predicates.model.SetValue;
+import com.contentgrid.thunx.predicates.model.ThunkExpression;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Data
+@EqualsAndHashCode
+@NoArgsConstructor
+@AllArgsConstructor
+public class JsonCollectionValueDto<T> implements JsonExpressionDto {
```

Comment:
I still think this is beter modelled as two separate classes, just like `ListValue` & `SetValue` in thunx-model.

- `JsonSetValueDto`
- `JsonArrayValueDto`

The reason is that you have separate implementations for every method anyways.